### PR TITLE
add core notes 2020-06-12

### PR DESCRIPTION
**Pull request by hudson-ayers**
### Pull Request Overview

This pull request adds notes from todays core wg meeting. Hopefully I represented everything correctly, there was a good amount of fast-paced discussion that was pretty technical.

### Testing Strategy

N/A

### TODO or Help Wanted

N/A

### Documentation Updated

- [x] Updated the relevant files in `/docs`

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

**Comment by ppannuto** [comment]
bors retry

File: doc/wg/core/notes/core-2020-06-12.md [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+of these things would be enforced behind a grant so you can't store ad-hoc
+process state in a capsule
+ * Amit: Sorry, what can you not do then?
+ * Brad: (explains again)
+ * Hudson: Brad is not dissenting, just saying he likes it for a different
+   reason
+ * Amit: Oh okay good, so we have consensus, so we should politely and
+   thankfully ask Guillaume to go ahead with a PR
+ * Leon: I have another approach, Can I open a PR with this other approach so
+   we can compare them both
```
Comment (gendx) [inline doc/wg/core/notes/core-2020-06-12.md:211]: @lschuermann please go ahead if you have some draft PR to experiment with this, and/or some design sketch to share on #1905! On my side, I likely won't have time to work on a PR proposal this week.

Comment (lschuermann) [reply doc/wg/core/notes/core-2020-06-12.md:211]: Thanks! I'll see what I can do, I don't know whether I'll make it this week either. Given that our proposals have only minor technical differences (from what I see the difference manifests itself rather in the usage semantics), if you have any idea for implementing i.e. the core kernel-held allow-table efficiently, I'd love to chat about that (e.g. on Slack).

**Pull request merged**